lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2566,21 +2592,26 @@ static SILFunction *getAssociatedFunction(
 namespace {
 class VJPEmitter final
     : public TypeSubstCloner<VJPEmitter, SILOptFunctionBuilder> {
+  friend class AdjointEmitter;
+
 private:
   /// The global context.
   ADContext &context;
 
   /// The original function.
-  SILFunction *original;
+  SILFunction *const original;
 
   /// The `[differentiable]` attribute.
-  SILDifferentiableAttr *attr;
+  SILDifferentiableAttr *const attr;
 
   /// The VJP function.
-  SILFunction *vjp;
+  SILFunction *const vjp;
 
-  /// The primal info.
-  std::unique_ptr<PrimalInfo> primalInfo = nullptr;
+  /// The adjoint function.
+  SILFunction *adjoint;
+
+  /// The pullback info.
+  std::unique_ptr<PullbackInfo> pullbackInfo = nullptr;
```

Comment:
hmm why is this still a unique ptr?

Reply:
Nice catch 🙂 
That will be fixed in https://github.com/apple/swift/pull/24911. Changing that in this PR was difficult for some reason.

Reply:
My mistake - there's no reason why `PullbackInfo` can't be stored as a normal value instead of `unique_ptr`. I forgot to apply that change in this PR, it'll be fixed in my next patch.